### PR TITLE
Serialize unleashed features.

### PR DESCRIPTION
**Pull request by dvc94ch**
So this fixes the problem of jsonish output for debugging. the entire dag-json thing should be reworked anyway and in my preference not actually read/write compliant json but something that looks like json. But I'm not particulary interested in doing this. Especially since everybody will start complaining again about specs and compatibility.

### CODE REVIEW

File: dag-json/src/codec.rs [hunk 1]
```diff
@@ -39,14 +39,15 @@ fn serialize<S: ser::Serializer>(ipld: &Ipld, ser: S) -> Result<S::Ok, S::Error>
             ser.collect_map(wrapped)
         }
         #[cfg(feature = "unleashed")]
-        Ipld::IntegerMap(_) => {
-            // TODO: how to return an error here?
-            unimplemented!()
+        Ipld::IntegerMap(map) => {
+            let wrapped = map.iter().map(|(key, ipld)| (key, Wrapper(ipld)));
+            ser.collect_map(wrapped)
```
Comment (rklaehn) [inline dag-json/src/codec.rs:44]: So this will produce json like ```{ 1: "foo", 2: "bar" }```?

File: dag-json/src/codec.rs [hunk 2]
```diff
@@ -39,14 +39,15 @@ fn serialize<S: ser::Serializer>(ipld: &Ipld, ser: S) -> Result<S::Ok, S::Error>
...
         }
         #[cfg(feature = "unleashed")]
-        Ipld::Tag(_, _) => {
-            // TODO: how to return an error here?
-            unimplemented!()
+        Ipld::Tag(tag, ipld) => {
+            let mut map = BTreeMap::new();
+            map.insert("/", (tag, Wrapper(ipld)));
+            ser.collect_map(map)
```
Comment (rklaehn) [inline dag-json/src/codec.rs:50]: So ```{ "/": (42, "thevalue") }```

**Review by rklaehn — APPROVED** [review]
Looks good, but it would be great to have tests that show what the result is.

File: dag-json/src/codec.rs (hunk 1 above)
Comment (dvc94ch) [reply dag-json/src/codec.rs:44]: This lets serde_json take care of it, I think this will result in string keys

File: dag-json/src/codec.rs (hunk 2 above)
Comment (dvc94ch) [reply dag-json/src/codec.rs:50]: this also lets serde take care of it, I assume it will use an array

**Comment by dvc94ch** [comment]
well, for now it's just for debugging and it's behind the unleashed flag, so the output doesn't really matter.

File: dag-json/src/codec.rs (hunk 1 above)
Comment (rklaehn) [reply dag-json/src/codec.rs:44]: serde_json will produce int keys for an int keyed map as far as I know. But that is what I want, so it is fine.

Comment (vmx) [reply dag-json/src/codec.rs:44]: The keys will likely be a string as JSON doesn't support numbers as keys for objects (only JavaScript does).

Comment (rklaehn) [reply dag-json/src/codec.rs:44]: Right. Forgot about that. Anyway, this just to have a way to produce debug output for arbitrary dag-cbor unleashed data, so it is fine.

**Pull request merged**